jetty-websocket/websocket-core/src/main/java/org/eclipse/jetty/websocket/core/FrameHandler.java

```diff
@@ -510,12 +526,24 @@ public Duration getIdleTimeout()
             return timeout==null ? Duration.ZERO : timeout;
         }
 
+        @Override
+        public Duration getWriteIdleTimeout()
+        {
+            return timeout==null ? Duration.ZERO : timeout;
```

Comment:
prefer to initialise to ZERO and then make sure never set to null.   Ie put the extra if in the setting path rather than the more frequent getting path

Reply:
initialising to zero here would break the `ConfigurationCustomizer`, when `customize()` is called we only want to customize on configuration which has been set on the `ConfigurationCustomizer`, so if we initialised this to ZERO then this timeout would always be set on the `CoreSession` potentially overwriting other values

Reply:
ahh do you mean initialise it to ZERO on the setter

I don't think actually think we need that if on the setting or the getting path, it seems incorrect either way, if you call `getWriteIdleTimeout()` and it returns ZERO because it has a value of null that is incorrect because it is not going to customize the session with a value of ZERO

Reply:
I mean that we should try to avoid an if in the getter as that is used much more frequently.    I'm fine with tri-stating using null as not set... but I don't think that applies here... only in the customizer.

Reply:
we could implement this behaviour on the setter for the getters for the timeouts as they return `Duration` which can be null, however for all the other methods this will result in NPE as they return primitive types, so I can do this it will just be a bit inconsistent with the rest of the methods